ChopChop/Models/Ingredients/Quantity.swift

```diff
@@ -0,0 +1,258 @@
+            guard difference >= 0 else {
+                throw QuantityError.negativeQuantity
+            }
+            return .volume(difference)
+        default:
+            throw QuantityError.differentTypes
+        }
+    }
+
+    /**
+     Returns the quantity scaled with a given factor.
+     - Throws:
+        - `QuantityError.negativeQuantity`: if the result is negative.
+     */
+    static func * (left: Quantity, right: Double) throws -> Quantity {
+        switch left {
+        case .count(let value):
+            let product = value * right
+            guard product >= 0 else {
+                throw QuantityError.negativeQuantity
+            }
+            return .count(product)
+        case .mass(let value):
+            let product = value * right
+            guard product >= 0 else {
+                throw QuantityError.negativeQuantity
+            }
+            return .mass(product)
+        case .volume(let value):
+            let product = value * right
+            guard product >= 0 else {
+                throw QuantityError.negativeQuantity
+            }
+            return .volume(product)
+        }
+    }
+
+    /**
+     Returns the quantity divided by a given factor.
+     - Throws:
+        - `QuantityError.divisionByZero`: if the given factor is 0.
+        - `QuantityError.negativeQuantity`: if the result is negative.
+     */
+    static func / (left: Quantity, right: Double) throws -> Quantity {
+        guard right != 0 else {
+            throw QuantityError.divisionByZero
+        }
+
+        switch left {
+        case .count(let value):
+            let quotient = value / right
+            guard quotient >= 0 else {
+                throw QuantityError.negativeQuantity
+            }
+            return .count(quotient)
+        case .mass(let value):
+            let quotient = value / right
+            guard quotient >= 0 else {
+                throw QuantityError.negativeQuantity
+            }
+            return .mass(quotient)
+        case .volume(let value):
+            let quotient = value / right
+            guard quotient >= 0 else {
+                throw QuantityError.negativeQuantity
+            }
+            return .volume(quotient)
+        }
+    }
+
+    static func += (left: inout Quantity, right: Quantity) throws {
+        let sum = try left + right
+        left = sum
+    }
+
+    static func -= (left: inout Quantity, right: Quantity) throws {
+        let difference = try left - right
+        left = difference
+    }
+
+    static func *= (left: inout Quantity, right: Double) throws {
+        let product = try left * right
+        left = product
+    }
+
+    static func /= (left: inout Quantity, right: Double) throws {
+        let quotient = try left / right
+        left = quotient
+    }
+}
+
+// MARK: - CustomStringConvertible
+extension Quantity: CustomStringConvertible {
+    var description: String {
```

Comment:
does this part display the quantity magnitude sensibly ie 50ml vs 0.05 litre

Reply:
Thanks for the suggestion, I'll add it in!